### ISSUE TITLE
Error CS0017 absent from squiggles and error-list "program has more than one entrypoint"

### ISSUE THREAD

**Issue by ljw1004** [dotnet/roslyn#12858]
**Version Used**: VS2015 Update 3

**Steps to Reproduce**:

1. File > New > C# > ConsoleApplication
2. Paste the code below
3. Build > BuildSolution

```csharp
class Program1 { static void Main() { } }
class Program2 { static void Main() { } }
```

**Expected Behavior**:

It should show a squiggle somewhere.

It should show CS0017 in the Error List.

**Actual Behavior**:

There is no squiggle, and nothing appears in the error list.

![cs0017](https://cloud.githubusercontent.com/assets/3316258/17307087/33f2dd74-57e8-11e6-8988-58fe699a503a.png)

**Comment by Pilchie** [comment]
This is fallout from https://github.com/dotnet/roslyn/pull/12195.

**Comment by Pilchie** [comment]
Tag @heejaechang do these need to be added to the list of things that aren't reported by the IDE or something?

**Comment by heejaechang** [comment]
@Pilchie right now, this is how we get compiler diagnostics

http://index/#Microsoft.CodeAnalysis/DiagnosticAnalyzer/CompilerDiagnosticAnalyzer.CompilationAnalyzer.cs,21

for error to show up, one of these 4 GetDiagnostics call we do on compiler should return diagnostics

http://index/#Microsoft.CodeAnalysis/DiagnosticAnalyzer/CompilerDiagnosticAnalyzer.CompilationAnalyzer.cs,21

**Comment by Pilchie** [comment]
As of #12195, the compiler will no longer return diagnostics related to the `Main()` method, if Full Solution Analysis is not enabled.

**Comment by Pilchie** [comment]
(I only see 3 calls there).

So what if none of those calls will return it?

**Comment by heejaechang** [comment]
@Pilchie 
semanticModel.GetSyntaxDiagnostics
SemanticModel.GetDeclarationDiagnostics
SemanticModel.GetMethodBodyDiagnostics
Compilation.GetDeclarationDiagnostics

then it won't be reported live. it will only reported from build.

**Comment by Pilchie** [comment]
I missed that there were two calls inside `AnalyzeSemanticModel`.

Don't we filter errors from the build from showing up in the error list if we think we're going to get them from the IDE analyzer?

**Comment by heejaechang** [comment]
@Pilchie yes. 
http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp/Compilation/CSharpCompilerDiagnosticAnalyzer.cs,13

if it is something only build can report but not live, it must be specified here.

**Comment by srivatsn** [comment]
Assigning to you @heejaechang to add this error to that list.
